lib/diagnostics_channel.js

```diff
@@ -23,11 +29,44 @@ const { triggerUncaughtException } = internalBinding('errors');
 
 const { WeakReference } = internalBinding('util');
 
+function decRef(channel) {
+  channel._weak.decRef();
+  if (channel._weak.getRef() === 0) {
```

Comment:
Maybe `decRef()` could return the value after decrementing so you don't need to call `getRef()`?

Reply:
Maybe? That would require changing the WeakReference type though. I could do that instead if you feel strongly about it. It would make the performance of `subscribe` and `unsubscribe` a tiny bit better, though that shouldn't be happening frequently to begin with. The API design scopes everything into the channel objects to decouple the lookup part of channel management from the publishing point, which is the hot part of the code. The parts _other_ than publishing should generally be very, very rare in comparison so performance is a lot less important there.

Reply:
It was fairly straightforward so I just made the change anyway.